app/controllers/FileChecksController.scala

```diff
@@ -119,16 +121,45 @@ class FileChecksController @Inject() (
         ).uncache()
       } else {
         if (isJudgmentUser) {
-          Ok(views.html.judgment.judgmentFileChecksProgress(consignmentId, reference, applicationConfig.frontEndInfo, request.token.name)).uncache()
+          if (applicationConfig.blockAutomateJudgmentTransfers) {
+            Ok(views.html.judgment.judgmentFileChecksProgress(consignmentId, reference, applicationConfig.frontEndInfo, request.token.name)).uncache()
+          } else {
+            val totalSleepTime = 3 * 60 * 1000 // Total sleep time in milliseconds (5 minutes)
+            Await.result(waitForFileChecksToBeCompleted(consignmentId, totalSleepTime), totalSleepTime.millis + 1.minute) // Adding a little buffer time
+            Redirect(routes.FileChecksResultsController.judgmentFileCheckResultsPage(consignmentId)).uncache()
+          }
         } else {
           Ok(views.html.standard.fileChecksProgress(consignmentId, reference, applicationConfig.frontEndInfo, request.token.name))
             .uncache()
         }
       }
-    }).recover { case exception: Exception =>
+    }).recover { case _: Exception =>
       Ok(views.html.uploadInProgress(consignmentId, reference, "Uploading your records", request.token.name, isJudgmentUser)).uncache()
     }
   }
+
+  private def waitForFileChecksToBeCompleted(consignmentId: UUID, totalSleepTime: Int)(implicit request: Request[AnyContent]): Future[Unit] = {
+    val interval = 5 * 1000 // Interval time in milliseconds (5 seconds)
+    val intervals = totalSleepTime / interval
+
+    def fileCheckProgress: Future[Boolean] = consignmentService
+      .fileCheckProgress(consignmentId, request.token.bearerAccessToken)
+      .map(consignment =>
+        consignment.fileChecks.checksumProgress.filesProcessed == consignment.totalFiles &&
+          consignment.fileChecks.ffidProgress.filesProcessed == consignment.totalFiles &&
+          consignment.fileChecks.antivirusProgress.filesProcessed == consignment.totalFiles
+      )
+    breakable {
```

Comment:
Feel free to ignore this as it may be a bit more unreadable. But to avoid using blocking with `await` this uses chaining with flatMap. It also introduces `future` with `blocking` for the sleep so that the execution context can manage thread pool resources more efficiently.

```    
def loop(remainingIntervals: Int): Future[Unit] = {
      if (remainingIntervals <= 0) {
        Future.unit
      } else {
        fileCheckProgress.flatMap { isCompleted =>
          if (isCompleted) {
            Future.unit
          } else {
            Future {
              blocking(Thread.sleep(interval))
            }.flatMap(_ => loop(remainingIntervals - 1))
          }
        }
      }
    }

loop(intervals)
```

Reply:
That's a good approach. I have used this. Thanks